src/otx/core/model/module/anomaly/anomaly_lightning.py

```diff
@@ -0,0 +1,450 @@
+        then It will call self._reset_prediction_layer() to reset the prediction layer. Which is not required.
+
+        This overrides the OTXModel's label_info setter.
+        """
+        if isinstance(value, list):
+            # value can be greater than 2 as datumaro returns all anomalous categories separately
+            self._label_info = LabelInfo(label_names=["Normal", "Anomaly"], label_groups=[["Normal", "Anomaly"]])
+        else:
+            self._label_info = value
+
+    def _extract_mean_scale_from_transforms(self, transforms: list[Transform]) -> None:
+        """Extract mean and scale values from transforms."""
+        for transform in transforms:
+            name = transform.__class__.__name__
+            if "Resize" in name:
+                self.input_size = transform.size * 2  # transform.size has value [size], so *2 gives (size, size)
+            elif "Normalize" in name:
+                self.mean_values = transform.mean
+                self.scale_values = transform.std
+
+    @property
+    def trainable_model(self) -> str | None:
+        """Use this to return the name of the model that needs to be trained.
+
+        This might not be the cleanest solution.
+
+        Some models have multiple architectures and only one of them needs to be trained.
+        However the optimizer is configured in the Anomalib's lightning model. This can be used
+        to inform the OTX lightning model which model to train.
+        """
+        return None
+
+    def setup(self, stage: str | None = None) -> None:
+        """Setup the model."""
+        super().setup(stage)  # type: ignore[misc]
+        if hasattr(self.trainer, "datamodule") and hasattr(self.trainer.datamodule, "config"):
+            if hasattr(self.trainer.datamodule.config, "test_subset"):
+                self._extract_mean_scale_from_transforms(self.trainer.datamodule.config.test_subset.transforms)
+            elif hasattr(self.trainer.datamodule.config, "val_subset"):
+                self._extract_mean_scale_from_transforms(self.trainer.datamodule.config.val_subset.transforms)
+
+    def configure_callbacks(self) -> list[Callback]:
+        """Get all necessary callbacks required for training and post-processing on Anomalib models."""
+        image_metrics = ["AUROC", "F1Score"]
+        pixel_metrics = image_metrics if self.task != TaskType.CLASSIFICATION else None
+        return [
+            _PostProcessorCallback(),
+            _MinMaxNormalizationCallback(),  # ModelAPI only supports min-max normalization as of now
+            _ThresholdCallback(threshold="F1AdaptiveThreshold"),
+            _MetricsCallback(
+                task=self.task,
+                image_metrics=image_metrics,
+                pixel_metrics=pixel_metrics,
+            ),
+        ]
+
+    def training_step(
+        self,
+        inputs: AnomalyClassificationDataBatch | AnomalySegmentationDataBatch | AnomalyDetectionDataBatch | dict,
```

Comment:
Would it be an idea to use a `TypeAlias` here, similar to how we do it in anomalib?